src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -202,10 +202,11 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
 
                 if (notRegularProperty)
                 {
-                    Binder.CheckFeatureAvailability(location,
-                                                    isReadOnly ? MessageID.IDS_FeatureReadonlyAutoImplementedProperties :
-                                                                 MessageID.IDS_FeatureAutoImplementedProperties,
-                                                    diagnostics);
+                    Binder.CheckFeatureAvailability(
+                        syntax,
+                        location,
```

Comment:
This seems the only call to `CheckFeatureAvailability` where `locationOpt` is passed. What's the difference of location between `syntax.GetLocation()` and `location` in this case? 
Could we use the location from the syntax and remove the `locationOpt` parameter?